### PR TITLE
refactor: improve type-strictness

### PR DESCRIPTION
**Pull request by jorgenengelsen**
## Why is this pull request needed?

This pull request is needed because of....

## What does this pull request change?

- Enable mypy strict mode
- Add types from 3rd party dependencies to mypy
- Add stronger typing

## Issues related to this change:

### CODE REVIEW

File: api/src/authentication/models.py [hunk 1]
```diff
@@ -15,11 +16,11 @@ def check_privilege(self, required_level: "AccessLevel") -> bool:
         return False
 
     @classmethod
-    def __get_validators__(cls):
+    def __get_validators__(cls):  # type:ignore
```
Comment (jorgenengelsen) [inline api/src/authentication/models.py:19]: Seems to be a mypy bug here. Did not accept any of its own suggestions, just went in circle with reccomendations. Tried with "Iterable", "Generator", ""AccessLevel"

File: api/src/features/todo/entities/todo_item.py [hunk 2]
```diff
@@ -8,13 +8,13 @@ class TodoItem:
...
 
     @classmethod
-    def from_dict(cls, dict_) -> "TodoItem":
+    def from_dict(cls, dict_: dict[str, str | bool]) -> "TodoItem":
         class_fields = {f.name for f in fields(cls)}
         if "_id" in dict_:
             dict_["id"] = dict_.pop("_id")
         data = {k: v for k, v in dict_.items() if k in class_fields}
-        return TodoItem(**data)
+        return TodoItem(**data)  # type:ignore
```
Comment (jorgenengelsen) [inline api/src/features/todo/entities/todo_item.py:20]: mypy did not like this for some reason, caused more typing problems upstream

File: api/src/features/todo/todo_feature.py [hunk 3]
```diff
@@ -57,8 +57,8 @@ def delete_todo_by_id(
 @router.get("", operation_id="get_all")
 def get_todo_all(
     user: User = Depends(auth_with_jwt), todo_repository: TodoRepositoryInterface = Depends(get_todo_repository)
-):
-    return get_todo_all_use_case(user_id=user.user_id, todo_repository=todo_repository)
+) -> list[GetTodoAllResponse]:
+    return get_todo_all_use_case(user_id=user.user_id, todo_repository=todo_repository)  # type: ignore
```
Comment (sebastianvitterso) [inline api/src/features/todo/todo_feature.py:61]: Why the `# type: ignore` here?

**Review by sebastianvitterso — APPROVED** [review]
Looks good!

File: api/src/features/todo/todo_feature.py (hunk 3 above)
Comment (jorgenengelsen) [reply api/src/features/todo/todo_feature.py:61]: I could not figure it out, Mypy insists it is returning None, but the function has return type `list[GetTodoAllResponse]`.

So i basicly gave up

**Pull request merged**